tasks/eva_3380/src/main/groovy/eva3380/eva3380_collect_colliding_ss_hashes.groovy

```diff
@@ -0,0 +1,168 @@
+    def customAssemblyReportPath = runProcess("find ${fastaDir} -maxdepth 3 -iname ".toString() +
+            "'${assembly}_assembly_report_custom.txt'| head -1".toString()).out.trim()
+    if (customAssemblyReportPath.equals("")) throw new Exception("Could not find assembly report for assembly $assembly!!")
+    return customAssemblyReportPath
+}
+
+// Copied verbatim from here: https://github.com/EBIvariation/eva-tasks/blob/27e4a1bda59fb30e8baabd2500119ae9ffacf05d/tasks/eva_3371/src/main/groovy/eva3371/eva3371_detect_unnormalized_indels.groovy#L154
+def getNormalizedDbsnpAndEvaSves = { ArrayList<Variant> variantsPostNorm,
+                                     EVADatabaseEnvironment prodEnv ->
+    def variantsPostNormByHash = variantsPostNorm.groupBy {
+        it.sourceEntries[0].getAttribute("SS_HASH")
+    }
+    def hashesToLookInProd = variantsPostNormByHash.keySet()
+    def normalizedDbsnpAndEvaSves =
+            [dbsnpSveClass, sveClass].collect {collectionClass ->
+                prodEnv.mongoTemplate.find(query(where("_id").in(hashesToLookInProd)),
+                        collectionClass).collect { oldSVE ->
+                    def variantPostNorm = variantsPostNormByHash.get(oldSVE.hashedMessage)[0]
+                    def updatedSV = new SubmittedVariant(oldSVE.referenceSequenceAccession,
+                            oldSVE.taxonomyAccession, oldSVE.projectAccession, oldSVE.contig,
+                            variantPostNorm.start, variantPostNorm.reference, variantPostNorm.alternate,
+                            oldSVE.clusteredVariantAccession, oldSVE.isSupportedByEvidence(),
+                            oldSVE.isAssemblyMatch(), oldSVE.isAllelesMatch(), oldSVE.isValidated(),
+                            oldSVE.createdDate)
+                    def updatedHash = new SubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction()).apply(updatedSV)
+                    // Crudely tack on the old SS hash to the start of the remappingId since that is the
+                    // least harmless String attribute that allows us to encode extra information
+                    // without messing up the object model's integrity
+                    // We must be careful with this however when using this info in EVA-3380 to update Indels
+                    return new SubmittedVariantEntity(oldSVE.accession, updatedHash, updatedSV, oldSVE.version,
+                            oldSVE.remappedFrom, oldSVE.remappedDate,
+                            oldSVE.hashedMessage + "_" + oldSVE.remappingId)
+                }
+            }
+    return normalizedDbsnpAndEvaSves
+}
+
+def getClashingSSRecordsFromDb = {List<SubmittedVariantEntity> svesFromVCF,
+                                  EVADatabaseEnvironment devEnv, EVADatabaseEnvironment prodEnv ->
+    def svesFromVcfGroupedByHash = svesFromVCF.groupBy{it.hashedMessage}
+    def clashingSvesInDevAndProd = [devEnv, prodEnv].collect{env ->
```

Comment:
> is it the normalised variants loaded in EVA-3371
Yes
> Though I guess that's what's in svesFromVCF
Not all variants from the VCF made it to the EVA-3371 output because of hash collisions among the normalized indels. So, there could be two SS (ss1 in the VCF and ss2 in the EVA3371 output in the DEV database) both sharing the same locus (post normalization) and also another SS ss3 in PROD. This script is to collect all of those colliding SS hashes so that we can decide on the proper merge actions to be taken.

Reply:
Got it - thank you!